libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts

```diff
@@ -65,28 +71,38 @@ export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
      * Subscribe to {@link scroll$} observable and emit {@link scrollEnd} event
      * when element scroll position has surpassed the offset.
      */
-    this.scroll$
-      .pipe(
-        takeUntil(this.ngUnsubscribe$),
-        debounceTime(INFINITE_SCROLL_DEBOUNCE),
-        filter(() => !this.disabled),
-        filter((scroll) => {
-          return (
-            scroll.elementHeight * (1 - this.offset) >=
-            scroll.distanceToViewBottom - scroll.viewHeight
-          );
-        })
-      )
-      .subscribe(() => {
-        this.scrollEnd.emit();
-      });
 
+    if (!this.listComponent.isVirtualScrollEnabled) {
+      this.scroll$
+        .pipe(
+          takeUntil(this.ngUnsubscribe$),
+          debounceTime(INFINITE_SCROLL_DEBOUNCE),
+          filter(() => !this.disabled),
+          filter((scroll) => {
+            return (
+              scroll.elementHeight * (1 - this.offset) >=
+              scroll.distanceToViewBottom - scroll.viewHeight
+            );
+          })
+        )
+        .subscribe(() => {
+          this.scrollEnd.emit();
+        });
+    }
+
+    const timeoutTime = !this.listComponent.isVirtualScrollEnabled
+      ? INSTANT
+      : VIRTUAL_SCROLL_INITIALZATION;
     setTimeout(() => {
       const ionContent: HTMLElement = this.elementRef.nativeElement.closest('ion-content');
       if (ionContent) {
-        // we run the 'ionScroll' event outside angular, as it would trigger change detection on each scroll
+        const getInnerScrollElm = () => ionContent.shadowRoot.querySelector('.inner-scroll');
```

Comment:
We need to listen to the scroll event on the .inner-scroll with virtual scroll, for some reason ionScroll doesn't trigger with virtual scroll

Reply:
Wouldn't it be better to replace our custom infinite scroll directive with Ionics own?

Reply:
Can we do this in a separate PR maybe? This works now.